### PR TITLE
feat(AWS Lambda): Supports Python3.10

### PR DESCRIPTION
**Pull request by t3yamoto**
This PR add support for Python3.10 on AWS Lambda.
Currently, the following warning is output when deploying, but this can be suppressed.

```
Warning: Invalid configuration encountered  
at 'provider.runtime': must be equal to one of the allowed values [dotnet6, dotnetcore3.1, go1.x, java11, java8, java8.al2, nodejs12.x, nodejs14.x, nodejs16.x, nodejs18.x, provided, provided.al2, python3.6, python3.7, python3.8, python3.9, ruby2.7]
Learn more about configuration validation here: http://slss.io/configuration-validation
```

Closes: #11921

### CODE REVIEW

**Review by GrahamCampbell — DISMISSED** [review]

**Review by nuc-ronniehyslop — APPROVED** [review]
I've tested deployment using --force and can confirm 3.10 deploys correctly to AWS as 3.10.

**Review by giuliano-oliveira — DISMISSED** [review]

**Review by Tsingis — DISMISSED** [review]

**Review by nuc-ronniehyslop — DISMISSED** [review]

**Comment by bartelemi** [comment]
Amazing can we get this merged and released? Python 3.10 has been long awaited and this is the last thing we need in place to start using it! 😄

On a side note, Python 2.7 and Python 3.6 are no longer supported by AWS (you can't create or update lambda functions anymore), so SLS could drop the support for these runtimes already.

**Comment by GrahamCampbell** [comment]
> so SLS could drop the support for these runtimes already.

Out of scope - should be another PR.

**Review by medikoo — DISMISSED** [review]
Thank you @t3yamoto. Sorry that it took that long, but I was on vacation, and needed to catch up afterwards

File: lib/plugins/aws/invoke-local/index.js [hunk 1]
```diff
@@ -253,7 +253,7 @@ class AwsInvokeLocal {
       );
     }
 
-    if (['python2.7', 'python3.6', 'python3.7', 'python3.8', 'python3.9'].includes(runtime)) {
+    if (['python2.7', 'python3.6', 'python3.7', 'python3.8', 'python3.9', 'python3.10'].includes(runtime)) {
```
Comment (giuliano-oliveira) [inline lib/plugins/aws/invoke-local/index.js:256]: Prettier is failing in the CI, this should make it work:
```suggestion
    if (
      ['python2.7', 'python3.6', 'python3.7', 'python3.8', 'python3.9', 'python3.10'].includes(
        runtime
      )
    ) {
```

Comment (t3yamoto) [reply lib/plugins/aws/invoke-local/index.js]: @giuliano-oliveira Thank you for the suggestion ! I should have followed the documentation below.
https://github.com/serverless/serverless/blob/main/CONTRIBUTING.md#verifying-prettier-formatting

**Review by nuc-ronniehyslop — APPROVED** [review]

**Comment by t3yamoto** [comment]
Three CI jobs are failing, but I don't seem to be authorized to check the details. Is there anything I can do?

**Review by medikoo — APPROVED** [review]
Thank you @t3yamoto 👍

**Comment by medikoo** [comment]
> Three CI jobs are failing, but I don't seem to be authorized to check the details. Is there anything I can do?

It was due to quirk on GitHub Actions side, otherwise build results are public

**Comment by giuliano-oliveira** [comment]
@medikoo Do we have a estimate for this change to be released?

**Comment by medikoo** [comment]
@giuliano-oliveira I'm finalizing reviews of the remaining PR's and it'll be released early next week. Sorry for the wait

**Comment by giuliano-oliveira** [comment]
No worries, thanks for the response!

**Pull request merged**